src/io/syncfile.c

```diff
@@ -433,11 +405,26 @@ static int resolve_open_mode(int *flag, const char *cp) {
     }
 }
 
+/*
+    Returns only the part of the open mode relevant to checking readonly/writeonly/readwrite checks
+    (Because O_RDONLY is 0, we can't simply use bitwise logic).
+ */
+static void resolve_open_mode_simple(int *flag, const char *cp) {
```

Comment:
Why the rw parameter and not just return the result?

Reply:
I've since removed this extra function and now set the simplified flag value directly in `resolve_open_mode`. It's still an rw parameter, but at least it has a reason to be one now :)